materialchips/src/main/ets/components/MainPage/ActionChip.ets

```diff
@@ -15,67 +15,188 @@
+
+    getImage(): string | Resource {
+      return this.img
+    }
+    setImage(img: string | Resource): Properties {
+      if (img != null && img != undefined) {
+        this.img = img;
+      } else {
+        this.img = "";
+      }
+      return this
+    }
+    getImageSize(): number {
+      return this.imgsize
+    }
+
+    setImageSize(imgsize: number): Properties {
+      if (imgsize != null && imgsize != undefined) {
+        this.imgsize = imgsize;
+      } else {
+        this.imgsize = 16;
+      }
+      return this
+    }
+    getTextSize(): number {
+      return this.textsize
+    }
+
+    setTextSize(textsize: number): Properties {
+      if (textsize != null && textsize != undefined) {
+        this.textsize = textsize;
+      } else {
+        this.textsize = 20;
+      }
+      return this
+    }
+    getBorderRadius(): number {
+      return this.borderradius
+    }
+
+    setBorderRadius(borderradius: number): Properties {
+      if (borderradius != null && borderradius != undefined) {
+        this.borderradius = borderradius;
+      } else {
+        this.borderradius = 8;
+      }
+      return this
+    }
+
+    getTextColor(): ResourceColor | string {
+      return this.textcolor
+    }
+
+    setTextColor(textcolor: ResourceColor | string): Properties {
+      this.textcolor = (textcolor != null && textcolor != undefined) ? textcolor : '#000000';
+      return this
+    }
+
+    getActiveTextColor(): ResourceColor | string {
+      return this.activetextcolor
+    }
+
+    setActiveTextColor(activetextcolor: ResourceColor | string): Properties {
+      this.activetextcolor = (activetextcolor != null && activetextcolor != undefined) ? activetextcolor : '#000000';
+      return this
+    }
+    getColor(): ResourceColor | string {
+      return this.color
+    }
+    setColor(color: ResourceColor | string): Properties {
+      this.color = (color != null && color != undefined) ? color : this.chipcolor;
+      return this
+    }
+    getChipColor(): ResourceColor | string {
+      return this.chipcolor
+    }
+
+    setChipColor(chipcolor: ResourceColor | string): Properties {
+      this.chipcolor = (chipcolor != null && chipcolor != undefined) ? chipcolor : '#000000';
+      return this
+    }
+
+    getActiveChipColor(): ResourceColor | string {
+      return this.activechipcolor
+    }
+
+    setActiveChipColor(activechipcolor: ResourceColor | string): Properties {
+      this.activechipcolor = (activechipcolor != null && activechipcolor != undefined) ? activechipcolor : '#000000';
+      return this
+    }
+
+    reset() {
+      this.textsize = 20
+
+      this.borderradius = 8
+      this.textcolor = '#000000'
+
+      this.activetextcolor = '#000000'
+      this.chipcolor = '#DCDCDC'
+
+      this.activechipcolor = '#800080'
+    }
+  }
+
+  export class Model extends Properties {
+    private text: string | Resource = undefined
+    constructor(text: string | Resource) {
```

Comment:
Include the chiptext also in parent class. Child class not required in this scenario. Do these changes for other chip types as well.